src/NServiceBus.Core.Tests/Performance/TimeToBeReceived/TimeToBeReceivedAttributeTests.cs

```diff
@@ -6,27 +6,82 @@
     public class TimeToBeReceivedAttributeTests
     {
         [Test]
-        public void Should_use_TimeToBeReceived_from_bottom_of_tree()
+        public void Should_use_TimeToBeReceived_from_bottom_of_tree_when_initialized()
         {
-            var mappings = new TimeToBeReceivedMappings(new[]{typeof(InheritedClassWithAttribute)},TimeToBeReceivedMappings.DefaultConvention);
+            var mappings = new TimeToBeReceivedMappings(new[]
+            {
+                typeof(InheritedClassWithAttribute)
+            }, TimeToBeReceivedMappings.DefaultConvention, true);
 
             TimeSpan timeToBeReceived;
 
-            Assert.True(mappings.TryGetTimeToBeReceived(typeof(InheritedClassWithAttribute),out timeToBeReceived));
+            Assert.True(mappings.TryGetTimeToBeReceived(typeof(InheritedClassWithAttribute), out timeToBeReceived));
             Assert.AreEqual(TimeSpan.FromSeconds(2), timeToBeReceived);
         }
 
         [Test]
-        public void Should_use_inherited_TimeToBeReceived()
+        public void Should_use_inherited_TimeToBeReceived_when_initialized()
         {
-            var mappings = new TimeToBeReceivedMappings(new[] { typeof(InheritedClassWithNoAttribute) }, TimeToBeReceivedMappings.DefaultConvention);
+            var mappings = new TimeToBeReceivedMappings(new[]
+            {
+                typeof(InheritedClassWithNoAttribute)
+            }, TimeToBeReceivedMappings.DefaultConvention, true);
 
             TimeSpan timeToBeReceived;
 
             Assert.True(mappings.TryGetTimeToBeReceived(typeof(InheritedClassWithNoAttribute), out timeToBeReceived));
             Assert.AreEqual(TimeSpan.FromSeconds(1), timeToBeReceived);
         }
 
+        [Test]
+        public void Should_throw_when_discard_before_received_not_supported_when_initialized()
+        {
+            Assert.That(() => new TimeToBeReceivedMappings(new[]
+            {
+                typeof(InheritedClassWithNoAttribute)
+            }, TimeToBeReceivedMappings.DefaultConvention, doesTransportSupportDiscardIfNotReceivedBefore: false), Throws.Exception.Message.StartWith("Messages with TimeToBeReceived found but the selected transport does not support this type of restriction"));
+        }
+
+        [Test]
+        public void Should_use_TimeToBeReceived_from_bottom_of_tree_when_tryget()
```

Comment:
naming nitpicking: "when_tryget" seems to point at the tested method. Shouldn't that follow some different convention like "TryGet_when_..._should" (I know we don't have any conventions, but this seems to be just another way on top of the hundreds we already have?)